ui/app/components/status-badge/index.ts

```diff
@@ -0,0 +1,59 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+interface StatusBadgeArgs {
+  state: number;
+  iconOnly: boolean;
+}
+
+export default class StatusBadge extends Component<StatusBadgeArgs> {
+  @tracked iconOnly = false;
+  @tracked state: number;
+
+  constructor(owner: any, args: StatusBadgeArgs) {
+    super(owner, args);
+    this.state = args.state;
+    this.iconOnly = args.iconOnly;
+  }
+
+  get statusClass() {
+    let { state } = this.args;
+    switch (state) {
+      case 4:
+        return 'partial';
+        break;
+      case 3:
+        return 'down';
+        break;
+      case 2:
+        return 'ready';
+        break;
+      case 1:
+        return 'alive';
+        break;
+      case 0:
+        return 'unknown';
```

Comment:
Forgive me for not knowing typescript (I think??) but is there a `case default` or something where we return unknown always?

Reply:
The API should behave this way too, but maybe in the case where something goes wrong the UI could handle it the same way

Reply:
@briancain You're right I should have a default case too